internal/app/app.go

```diff
@@ -1445,6 +1449,25 @@ func (app *App) repairSlaveOfflineMode(host string, node *mysql.Node, state *Nod
 					host, *state.SlaveState.ReplicationLag, app.config.OfflineModeEnableLag)
 			}
 		}
+		// gradual transfer of permanently broken nodes to offline
+		lastShutdownNodeTime, err := app.GetLastShutdownNodeTime()
+		if err != nil {
+			app.logger.Errorf("repair: failed to get last shutdown node time: %s", err)
+			return
+		}
+		setOfflineIsPossible := time.Since(lastShutdownNodeTime) < app.config.OfflineModeEnableInterval
+		if result, _ := state.IsReplicationPermanentlyBroken(); result && !state.IsOffline && setOfflineIsPossible {
```

Comment:
Could we calculate IsReplicationPermanentlyBroken only once at the top of this method?